gadget.yaml

```diff
@@ -26,3 +26,18 @@ volumes:
             target: EFI/boot/bootx64.efi
           - source: grub.cfg
             target: EFI/ubuntu/grub.cfg
+      - name: Recovery
+        filesystem: vfat
+        # FIXME: what type to use?
+        type: EF,C12A7328-F81F-11D2-BA4B-00A0C93EC93B
```

Comment:
Hmmm.... That is ESP type, and you appear to have ESP like content too. I don't believe it is allowed as per UEFI spec to have multiple ESP partitions.

Recovery must be booted from the system-boot partition.... Or you need chainload from there....

Reply:
Thanks, I was suspecting this, I will change things.

Reply:
I changed this to have a single ESP partition now - I think this makes it easier and more reliable. However in my reading I could not find anything about a limit of ESP partitions. I don't know if the boot order is defined if there are multiple ones so I think its wise to limit ourselfs to just one (especially since we don't need multiple ones).